ecommerce/serializers.py

```diff
@@ -511,6 +517,23 @@ def _fetch_and_validate_product(self, items):
             ) from exc
         return product, program_run
 
+    def _validate_product_internal(self, product_content_obj):
+        """
+        Check if the product being bought is an external product, We don't sell external products
+
+        Args:
+            product_content_obj (Course, CourseRun, Program or ProgramRun): Ideally, there should only be
+            CourseRun or Program but it could be either of those since we can add these as content types in products.
+        """
+        course_or_program = (
+            getattr(product_content_obj, "course", None)
+            or getattr(product_content_obj, "program", None)
+            or product_content_obj
+        )
+
+        if course_or_program.is_external:
+            raise ValidationError("We're sorry, This product cannot be purchased.")
```

Comment:
Is it possible for the end user to see this error? 

If so, I think it would help to provide a bit more context and a call to action:

```suggestion
            raise ValidationError("We're sorry, This product cannot be purchased on this web site. Please contact customer support for more information.")
```

If possible, include the email for support. It can be text; it doesn't need to be an active link.

Reply:
fyi @cachob

Reply:
>If possible, include the email for support. It can be text; it doesn't need to be an active link.

Think that should be possible, Will take a look.

Reply:
@pdpinch Update on this.

Instead of just changing it for external course error messages I've changed it for other error messages too when there is no basket item because this felt like a robust solution instead of just doing this for external courses. 

Here is how this will look. Let me know what you feel about it.

**For external products**
<img width="1244" alt="image" src="https://user-images.githubusercontent.com/34372316/228540266-c429d16c-5d52-4494-b40e-aa340efc6e96.png">


**For other types of errors when there is no product in the basket**
<img width="1339" alt="image" src="https://user-images.githubusercontent.com/34372316/228540080-569b1c9c-a63d-40f9-9c28-376c4603059a.png">

Reply:
Looks great, thanks.